### DISCUSSION TITLE
Node.js version compatibility

### DISCUSSION THREAD

**Discussion by henryqdineen** [reactwg/react-compiler#15] [General Questions] [answered]
Should the Node.js version compatibility for React Compiler be documented somewhere? Is it already? I see that there is no `engines.node` configuration i the `package.json` but one of the dependencies (`zod-validation-error`) does expect `>=18.0.0`.

We are currently using an older version of Node.js with webpack and running into issues with the compiler's usage of `Array.prototype.at` (supported in `>= 16.6.0`). It does work with a polyfill but it would be nice to know the official Node.js version compatibility story.

**Comment by josephsavona** [comment, marked as answer]
Thanks for posting! This is a good callout, we should document the minimum node version and set `engines.node` in package.json. A min version of 16.6 seems very reasonable at this point, we could probably be more aggressive and require something newer though.
